src/main/kotlin/com/github/blarc/ai/commits/intellij/plugin/settings/clients/ernie/ErnieClientService.kt

```diff
@@ -0,0 +1,60 @@
+package com.github.blarc.ai.commits.intellij.plugin.settings.clients.ernie
+
+import com.github.blarc.ai.commits.intellij.plugin.AICommitsUtils.getCredentialAttributes
+import com.github.blarc.ai.commits.intellij.plugin.AICommitsUtils.retrieveToken
+import com.github.blarc.ai.commits.intellij.plugin.notifications.Notification
+import com.github.blarc.ai.commits.intellij.plugin.notifications.sendNotification
+import com.github.blarc.ai.commits.intellij.plugin.settings.clients.LLMClientService
+import com.intellij.ide.passwordSafe.PasswordSafe
+import com.intellij.openapi.components.Service
+import com.intellij.openapi.components.service
+import com.intellij.util.text.nullize
+import dev.langchain4j.model.chat.ChatLanguageModel
+import dev.langchain4j.model.qianfan.QianfanChatModel
+import kotlinx.coroutines.CoroutineScope
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.launch
+
+@Service(Service.Level.APP)
+class ErnieClientService(private val cs: CoroutineScope) : LLMClientService<ErnieClientConfiguration>(cs) {
+
+    companion object {
+        @JvmStatic
+        fun getInstance(): ErnieClientService = service()
+    }
+
+    override suspend fun buildChatModel(client: ErnieClientConfiguration): ChatLanguageModel {
+        val apiKey = client.apiKey.nullize(true) ?: retrieveToken(client.apiKeyId)?.toString(true)
+        val secretKey = client.secretKey.nullize(true) ?: retrieveToken(client.secretKeyId)?.toString(true)
+
+        val builder = QianfanChatModel.builder()
+            .baseUrl(client.host)
+            .apiKey(apiKey)
+            .secretKey(secretKey)
+            .modelName(client.modelId)
+            .temperature(client.temperature.toDouble())
+        if (client.modelId == "ERNIE-Speed-128K"){
```

Comment:
Could we maybe use `client.modelId.lowercase()` and handle all model ids this way?

Reply:
> lowercase

Indeed, we should use lowercase to avoid matching issues caused by case inconsistency. However, I have a better idea: we can do this:

```
if (client.modelId == QianfanChatModelNameEnum.ERNIE_SPEED_128K.modelName){
    builder.endpoint("ernie-speed-128k")
}
```

You can find QianfanChatModelNameEnum.ERNIE_SPEED_128K [here](https://github.com/langchain4j/langchain4j/blob/0.32.0/langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/QianfanChatModelNameEnum.java#L12).

Reply:
Taking another look, this seems so weird though... Why do you need to send the endpoint to `ernie-speed-128k` just for this model? What does this endpoint attribute represent? Why is this not handled by the builder when passing the `.modelName(client.modelId)` ?

Reply:
Is this a bug in https://github.com/langchain4j/langchain4j/blob/0.32.0/langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/QianfanChatModelNameEnum.java#L12 ? And this if statement is a workaround?

If yes, please add a comment explaining this, so we'll know to remove it after the bug is fixed.

Reply:
I found your PR on langchain4j https://github.com/langchain4j/langchain4j/pull/1426. So please just add a comment that explains why this if statement is needed. Thanks :pray: .

Reply:
This is a bug. You can see the latest [QianfanChatModelNameEnum.java](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/QianfanChatModelNameEnum.java#L12), where it has already been fixed. However, the latest version of langchain4j remains 0.32.0.

In fact, when creating the [QianfanChatModel](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/QianfanChatModel.java#L123), no modelId or similar information was passed to the builder. Different models are distinguished through endpoint, as referenced [here](https://github.com/langchain4j/langchain4j/blob/b047fc44935e596df5dff282d69168b88b8d21f2/langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/client/QianfanApi.java#L15). Inside QianfanChatModel, the endpoint is swapped based on modelName, as can be seen [here](https://github.com/langchain4j/langchain4j/blob/main/langchain4j-qianfan/src/main/java/dev/langchain4j/model/qianfan/QianfanChatModel.java#L73). From the code, it appears that each endpoint in the QianfanChatModelNameEnum enum should be unique. However, in reality, there are two 'completions', which causes the free ERNIE-Speed-128K to eventually request the paid ERNIE-Bot.

You can refer to the documentation for the Qianfan large models:
https://cloud.baidu.com/doc/WENXINWORKSHOP/s/jlil56u11
https://cloud.baidu.com/doc/WENXINWORKSHOP/s/6ltgkzya5
Search for rpc/2.0/ai_custom/v1/wenxinworkshop/chat.

Reply:
So if langchain4j releases a new version, that judgment will no longer exist, but it does not exist now.